flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/SpeculativeScheduler.java

```diff
@@ -367,9 +373,13 @@ public void notifySlowTasks(Map<ExecutionVertexID, Collection<ExecutionAttemptID
             }
         }
 
-        executionDeployer.allocateSlotsAndDeploy(
-                newSpeculativeExecutions,
-                executionVertexVersioner.getExecutionVertexVersions(verticesToDeploy));
+        try {
```

Comment:
This change does not fully solve the problem.
The underlying reason is that `SpeculativeScheduler#notifySlowTasks()` is not invoked from JobMaster RPC, but instead from a scheduled executor. e.g. if exception happens in `blockSlowNodes()`, similar problem can happen too.
Given that this kind of exceptions are unexpected, i.e. bugs, we can simply use `FutureUtils.assertNoException()` to handle it. A reference can be `DefaultScheduler#restartTasksWithDelay()`.